lms/services/d2l_api/client.py

```diff
@@ -117,6 +117,32 @@ def course_table_of_contents(self, org_unit) -> List[dict]:
         )
         return D2LTableOfContentsSchema(response).parse().get("modules", [])
 
+    def public_url(self, org_unit, file_id) -> str:
+        """
+        Return the URL to download the given file.
+
+        As opposed to other LMS that return a one time signed URL that we can then pass along to via
+        D2L requires us to send the API authentication header to get the file contents.
+
+        To make sure the API authentication header is not expired we'll make a API
+        request here so if it needs to be refreshed we follow the standard token refresh procedure.
+
+        https://docs.valence.desire2learn.com/res/content.html#get--d2l-api-le-(version)-(orgUnitId)-content-topics-(topicId)
+        https://docs.valence.desire2learn.com/res/content.html#get--d2l-api-le-(version)-(orgUnitId)-content-topics-(topicId)-file
+        """
+
+        # We don't need the data from this call.
+        # We are only interested on the potential side effect of needing
+        # a new access token and/or refreshing an existing one.
+        _ = self._api.request(
```

Comment:
This is a bit of a hack but the whole setup is (sending the header etc).

Reply:
Is this because we're going to be sending our access token to Via (in the new secret headers), and we therefore need to make sure that the access token is still valid first?